app/SpecialDate.php

```diff
@@ -217,4 +217,23 @@ public function setToContact(Contact $contact)
         $this->contact_id = $contact->id;
         $this->save();
     }
+
+    public function getDeathAge()
+    {
+        if (is_null($this->date)) {
+            return;
+        }
+
+        if ($this->is_year_unknown) {
+            return;
+        }
+
+        $contact = Contact::findorfail($this->contact_id);
```

Comment:
`findOrFail`

Reply:
Dismiss last comment: you don't have to find the contact. You can access the contact by using `$this->contact` as they are linked with a `belongsTo` relationship.